src/vs/platform/environment/common/environment.ts

```diff
@@ -51,6 +51,7 @@ export interface IEnvironmentService {
 	appSettingsPath: string;
 	appKeybindingsPath: string;
 
+	isBackupEnabled: boolean;
```

Comment:
@Tyriar not so sure about introducing such a specific option to the environment service. keep in mind that it is being used in a lot of environments, including the CLI when managing extensions. I think a better approach is to introduce an `isExtensionDevelopment` flag that we can use to replace all our checks for `extensionDevelopmentPath` throughout the code.